### PR TITLE
CMR-5944 ISO-MENDS UMM-C Translation.

### PR DESCRIPTION
**Pull request by daniel-zamora**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj [hunk 1]
```diff
@@ -73,7 +73,21 @@
...
+                                                                           :YDimension 0.5
+                                                                           :Unit "Decimal Degrees"}
+                                                                          {:HorizontalResolutionProcessingLevelEnum "Gridded"
+                                                                           :XDimension 0.2
+                                                                           :Unit "Kilometers"}]
+                                              :Description "ResolutionAndCoordinateSystem Description value."
+                                              :GeodeticModel {:HorizontalDatumName "North American Datum 1983"
+                                                              :EllipsoidName "GRS 1980"
+                                                              :SemiMajorAxis 6378137
+                                                              :DenominatorOfFlatteningRatio 298.257222101}}}
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj]: Nitpick: fix indentation here

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj [hunk 2]
```diff
@@ -43,6 +44,49 @@
                 (:EastBoundingCoordinate  rect)
                 (:SouthBoundingCoordinate rect))))))
 
+(defn- remove-whitespace
+  [value]
+  (when value
+    (-> (clojure.string/split value #"\s")
+        clojure.string/join)))
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj]: This feels like a function that we would have in a utility namespace. If not, maybe move it there?

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj [hunk 3]
```diff
@@ -114,14 +115,190 @@
...
+        :MinimumXDimension (when (seq minimum-xdimension)
+                             (read-string minimum-xdimension))
+        :MaximumXDimension (when (seq maximum-xdimension)
+                             (read-string maximum-xdimension))
+        :ScanDirection (when (seq scan-direction)
+                         scan-direction)
+        :ViewingAngleType (when (seq viewing-angle-type)
+                            viewing-angle-type)
+        :HorizontalResolutionProcessingLevelEnum (when (seq horizontal-resolution-processing-level-enum)
+                                                   horizontal-resolution-processing-level-enum)}))))
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj]: Nitpick: is there a way to split the `let` above into a few smaller pieces? Some functions like `get-x-dimension`, `get-y-dimension`, etc might make this more readable.

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj (hunk 2 above)
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj]: change `clojure.string` to `string`.

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj [hunk 4]
```diff
@@ -92,6 +136,75 @@
          (map (partial umm-s/set-coordinate-system coordinate-system))
          (map gmd/encode))))
 
+(defn- resolution-and-coordinates-map->string
+  "Converts map to parsable string for ISO format."
+  [rcmap]
+  (when-let [map-keys (keys rcmap)]
+   (let [key-val-strings (map #(str (name %)  ": " (get rcmap %)) map-keys)]
+     (clojure.string/join " " key-val-strings))))
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/spatial.clj]: change `clojure.string` to `string`.

**Pull request merged**